framework/src/problems/FEProblemBase.C

```diff
@@ -3506,6 +3506,11 @@ FEProblemBase::prepareMaterials(SubdomainID blk_id, const THREAD_ID tid)
 
   setActiveElementalMooseVariables(needed_moose_vars, tid);
   setActiveMaterialProperties(needed_mat_props, tid);
```

Comment:
does need_mat_props contain all the properties found by dependency resolution?

Reply:
Based on the screen output, it does. One thing could be the case is that even a material property is not used on element but just used by a boundary, the material property will be marked as needed on element.